filter/impl/token_filter.go

```diff
@@ -0,0 +1,70 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package impl
+
+import (
+	"strings"
+)
+
+import (
+	perrors "github.com/pkg/errors"
+)
+
+import (
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/extension"
+	"github.com/apache/dubbo-go/filter"
+	"github.com/apache/dubbo-go/protocol"
+)
+
+const (
+	token = "token"
+)
+
+func init() {
+	extension.SetFilter(token, GetTokenFilter)
+}
+
+type TokenFilter struct{}
+
+func (tf *TokenFilter) Invoke(invoker protocol.Invoker, invocation protocol.Invocation) protocol.Result {
+	invokerTkn := invoker.GetUrl().GetParam(constant.TOKEN_KEY, "")
+	if len(invokerTkn) > 0 {
+		attachs := invocation.Attachments()
+		if len(attachs) > 0 {
+			remoteTkn, exist := attachs[constant.TOKEN_KEY]
+			if exist {
+				if strings.EqualFold(invokerTkn, remoteTkn) {
```

Comment:
if remoteTkn, exist := attachs[constant.TOKEN_KEY];exist&&strings.EqualFold(invokerTkn,remoteTkn){
}

Reply:
"remoteTkn, exist := attachs[constant.TOKEN_KEY]" should be one line, that will be clear.

Reply:
@hxmhlt done